src/extra/AutoUi/Collection/index.tsx

```diff
@@ -0,0 +1,385 @@
+// Make table handle tags better so we don't have to filter it out from the schema.
+// Specifically for the table, how should we treat columns that are based on multiple schema fields, shall we have some rule around that? Eg. the release "deployment" column combines "status" and "source" and it does appear more compact than having a separate column just to say it is a cloud build or a balena deploy one.
+// How shall we handle batch edits (especially when the data differs between entries)? We already do this for HUP but it is all custom code, we need to generalize it.
+// SDK's CRUD should all have the same function signature (eg. apiKey.create doesnt take a single object, but two separate fields).
+
+const HeaderGrid = styled(Flex)`
+	margin-left: -4px;
+	margin-right: -4px;
+
+	> * {
+		margin-left: 4px;
+		margin-right: 4px;
+	}
+`;
+
+const getSelectedItems = <T extends BaseResource<T>>(
+	newItems: T[],
+	selectedItems: T[],
+) => {
+	if (!size(selectedItems)) {
+		return selectedItems;
+	}
+	// update the selections
+	selectedItems = intersectionBy(newItems, selectedItems, 'id');
+	return selectedItems;
+};
+
+export interface ActionData<T> {
+	action: AutoUIAction<T>;
+	schema: JSONSchema;
+	affectedEntries?: T[];
+}
+
+export interface CollectionProps<T> {
+	/** Model is the property that describe the data to display with a JSON structure */
+	model: AutoUIModel<T>;
+	/** Array of data to display */
+	data: T[] | undefined;
+	/** Formats are custom widgets to render in the table cell. The type of format to display is described in the model. */
+	formats?: Format[];
+	/** Actions is an array of actions applicable on the selected items */
+	actions?: Array<AutoUIAction<T>>;
+	/** The sdk is used to pass the method to handle tags when added removed or updated */
+	sdk?: {
+		tags?: ResourceTagModelService;
+		refresh?: () => void;
+	};
+	/** Dictionary of {[column_property]: customFunction} where the customFunction is the function to sort data on column header click */
+	customSort?: Dictionary<(a: T, b: T) => void>;
+	/** Pass a component to display each data item in that component instead of a table */
+	cardRenderer?: AutoUIContext<T>['cardRenderer'];
+	// TODO: Ideally the base URL is autogenerated, but there are some issues with that currently (eg. instead of application we have apps in the URL)
+	/** Redirect on row click */
+	getBaseUrl?: (entry: T) => string;
+	/** Event emitted on row click */
+	onRowClick?: (
+		entry: T,
+		event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+	) => void;
+	t: any;
```

Comment:
I think we now have a type for this right?

Reply:
just created a type